torch/csrc/distributed/c10d/init.cpp

```diff
@@ -3216,7 +3219,16 @@ for details.
           // shouldn't leak because of allocation in strdup.
           [](ncclConfig_t& self, const char* tmp) {
             self.netName = strdup(tmp);
-          });
+          })
+      .def(
+          "__copy__",
+          [](const ncclConfig_t& self) { return ncclConfig_t(self); })
+      .def(
+          "__deepcopy__",
+          [](const ncclConfig_t& self, const py::dict& memo) {
+            return ncclConfig_t(self);
+          },
+          py::arg("memo"));
```

Comment:
I introduce the (deep)copy methods because in our application we need to activate the new flag only in a subgroup, but we want to retrieve the rest of the config from the root PG, hence we need to make a copy of it in oder to avoid modifying it in-place.